README.md

```diff
@@ -391,6 +391,36 @@ Advanced examples of using this trace event mechanism are:
 For more information, for example on how to define and register trace event consumers, please refer
 to the documentation in the `Tracer` class.
 
+## Module: Uid
+
+Generic immutable unique ID class. Really just an abstraction of UUIDs. Used to unique
+identify things. No 2 Uid objects shall be equal unless they are the same instance of the
+Uid class or their underlying UUIDs are equal.
+
+The class is a generic type to allow creating typesafe Uid's, like `Uid<Message>` or `Uid<Person>`.
+The class represents UUID as Strings internally, to avoid loads of UUID to String conversions
+all the time. This makes the class considerably faster in use than the regular Java UUID class.
+
+Examples:
+```kotlin
+val messageId = Uid<Message>()    // Creates a new unique message ID.
+val personId = Uid<Person>()      // Creates a new unique person ID.
+
+if (messageId == personId) {...}  // <-- Not allowed, the IDs are type safe.
```

Comment:
Is this syntactically not allowed or something? `if (other !is Uid<*>) return false` doesn't catch this and we're not using reified generics so it's not trivial to ensure it on a logic level.

Reply:
text is incorrect. Equality check is allowed, but assignment is not. Replaced text in README.